### PR TITLE
Remove type check function for repeated fields

### PR DESCRIPTION
**Pull request by szakarias**
Fixes https://github.com/dart-lang/protobuf/issues/201

### CODE REVIEW

File: protobuf/lib/src/protobuf/builder_info.dart [hunk 1]
```diff
@@ -113,13 +113,24 @@ class BuilderInfo {
...
+      [CreateBuilderFunc subBuilder,
+      ValueOfFunc valueOf,
+      List<ProtobufEnum> enumValues]) {
+    assert(_isGroupOrMessage(fieldType) || _isEnum(fieldType));
+    addRepeated<T>(tagNumber, name, fieldType, _checkNotNull, subBuilder,
+        valueOf, enumValues);
+  }
+
+  // Deprecated, use [pc] instead.
+  @deprecated
```
Comment (sigurdm) [inline protobuf/lib/src/protobuf/builder_info.dart]: I think there is a deprecated constructor that takes the description as an argument https://api.dartlang.org/stable/2.1.0/dart-core/Deprecated-class.html

File: protobuf/lib/src/protobuf/field_type.dart [hunk 2]
```diff
@@ -67,7 +67,7 @@ class PbFieldType {
 
   // Closures commonly used by initializers.
   static String _STRING_EMPTY() => '';
-  static List<int> _BYTES_EMPTY() => new PbList<int>(check: _checkInt);
+  static List<int> _BYTES_EMPTY() => new PbList<int>();
```
Comment (sigurdm) [inline protobuf/lib/src/protobuf/field_type.dart]: I think this should never have been a pblist.

File: protobuf/lib/src/protobuf/pb_list.dart [hunk 3]
```diff
@@ -48,6 +48,7 @@ class PbList<E> extends PbListBase<E> {
 
   PbList.from(List from) : super._from(from);
 
+  @deprecated
```
Comment (sigurdm) [inline protobuf/lib/src/protobuf/pb_list.dart]: Also here add explanation

**Comment by sigurdm** [comment]
cc @mkustermann this removes one of the closures from GeneratedMessage instances.

File: protobuf/lib/src/protobuf/builder_info.dart (hunk 1 above)
Comment (sigurdm) [reply protobuf/lib/src/protobuf/builder_info.dart]: You can also mention in the docstring that the check is ignored.

**Comment by szakarias** [comment]
PTAL

File: protobuf/lib/src/protobuf/pb_list.dart [hunk 4]
```diff
@@ -325,12 +325,4 @@ abstract class PbListBase<E> extends ListBase<E> {
     }
     _wrappedList.length = newLength;
   }
-
-  void _validate(E val) {
```
Comment (sigurdm) [inline protobuf/lib/src/protobuf/pb_list.dart]: ahhh...

File: protoc_plugin/lib/extension_generator.dart [hunk 5]
```diff
@@ -84,9 +84,10 @@ class ExtensionGenerator {
           ' \'$name\', ${_field.number}, ${_field.typeConstant}');
       if (type.isMessage || type.isGroup) {
         out.println(
-            ', $_protobufImportPrefix.checkNotNull, $dartType.create);');
+            ', $_protobufImportPrefix.getCheckFunction(${_field.typeConstant}), $dartType.create);');
       } else if (type.isEnum) {
-        out.println(', $_protobufImportPrefix.checkNotNull, null, '
```
Comment (sigurdm) [inline protoc_plugin/lib/extension_generator.dart]: remember to make checkNotNull private again.

File: protoc_plugin/pubspec.yaml [hunk 6]
```diff
@@ -10,7 +10,7 @@ environment:
 dependencies:
   fixnum: ^0.10.5
   path: ^1.0.0
-  protobuf: ^0.13.2
+  protobuf: ^0.13.3
```
Comment (sigurdm) [inline protoc_plugin/pubspec.yaml]: 0.13.4?

**Pull request merged**